contracts/contracts/strategies/NativeStaking/ValidatorRegistrator.sol

```diff
@@ -0,0 +1,349 @@
+        external
+        onlyRegistrator
+        whenNotPaused
+    {
+        uint256 requiredETH = validators.length * 32 ether;
+
+        // Check there is enough WETH from the deposits sitting in this strategy contract
+        require(
+            requiredETH <= IWETH9(WETH).balanceOf(address(this)),
+            "Insufficient WETH"
+        );
+        require(
+            activeDepositedValidators + validators.length <= MAX_VALIDATORS,
+            "Max validators reached"
+        );
+
+        require(
+            stakeETHTally + requiredETH <= stakeETHThreshold,
+            "Staking ETH over threshold"
+        );
+        stakeETHTally += requiredETH;
+
+        // Convert required ETH from WETH
+        IWETH9(WETH).withdraw(requiredETH);
+        _wethWithdrawnAndStaked(requiredETH);
+
+        /* 0x01 to indicate that withdrawal credentials will contain an EOA address that the sweeping function
+         * can sweep funds to.
+         * bytes11(0) to fill up the required zeros
+         * remaining bytes20 are for the address
+         */
+        bytes memory withdrawalCredentials = abi.encodePacked(
+            bytes1(0x01),
+            bytes11(0),
+            address(this)
+        );
+
+        // For each validator
+        for (uint256 i = 0; i < validators.length; ++i) {
+            bytes32 pubKeyHash = keccak256(validators[i].pubkey);
+            VALIDATOR_STATE currentState = validatorsStates[pubKeyHash];
+
+            require(
+                currentState == VALIDATOR_STATE.REGISTERED,
+                "Validator not registered"
+            );
+
+            IDepositContract(BEACON_CHAIN_DEPOSIT_CONTRACT).deposit{
+                value: 32 ether
+            }(
+                validators[i].pubkey,
+                withdrawalCredentials,
+                validators[i].signature,
+                validators[i].depositDataRoot
+            );
+
+            validatorsStates[pubKeyHash] = VALIDATOR_STATE.STAKED;
+
+            emit ETHStaked(
+                pubKeyHash,
+                validators[i].pubkey,
+                32 ether,
+                withdrawalCredentials
+            );
+        }
+        // save gas by changing this storage variable only once rather each time in the loop.
+        activeDepositedValidators += validators.length;
+    }
+
+    // slither-disable-end reentrancy-eth
+
+    /// @notice Registers a new validator in the SSV Cluster.
+    /// Only the registrator can call this function.
+    /// @param publicKey The public key of the validator
+    /// @param operatorIds The operator IDs of the SSV Cluster
+    /// @param sharesData The validator shares data
+    /// @param ssvAmount The amount of SSV tokens to be deposited to the SSV cluster
+    /// @param cluster The SSV cluster details including the validator count and SSV balance
+    // slither-disable-start reentrancy-no-eth
+    function registerSsvValidator(
```

Comment:
No change needed, just a note. I see that each of these SSV methods can only operate on one validator at a time. That may be somewhat less gas efficient, since I think we could do multiple submits on one cluster data.

However, I'm good keeping this as is for launch and we can tweak as needed later if active use shows it would matter.

Reply:
with the recent change to the P2P API introducing a delay between registering the SSV validator and getting the deposit data, which incudes the signature, I think we need to change `registerSsvValidator` to register multiple validators. I'll do this in a separate PR